src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationReader.java

```diff
@@ -0,0 +1,112 @@
+package se.liu.ida.hefquin.engine.wrappers.lpgwrapper.impl;
+
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ResourceRequiredException;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.ResIterator;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.riot.RDFDataMgr;
+import org.apache.jena.vocabulary.RDF;
+import se.liu.ida.hefquin.vocabulary.LPG2RDF;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class LPG2RDFConfigurationReader {
+
+    public static LPG2RDFConfigurationReader instance = new LPG2RDFConfigurationReader();
+
+    protected LPG2RDFConfigurationReader(){}
+
+    public static LPG2RDFConfigurationImpl readFromFile( final String filename ) {
+        return instance.parseLPG2RDFConf(filename);
+    }
+
+    public static LPG2RDFConfigurationImpl readFromModel(final Model lpg2Rdf ) {
+        return instance.parseLPG2RDFConf(lpg2Rdf);
+    }
+
+    public LPG2RDFConfigurationImpl parseLPG2RDFConf( final String filename ) {
+        final Model lpg2Rdf = RDFDataMgr.loadModel(filename);
+        return parseLPG2RDFConf(lpg2Rdf);
+    }
+
+    public LPG2RDFConfigurationImpl parseLPG2RDFConf(final Model lpg2Rdf ) {
+        final NodeMapping nodeMapping;
+        final String label;
+
+        final ResIterator lpg2rdfConfigs = lpg2Rdf.listResourcesWithProperty(RDF.type, LPG2RDF.LPGtoRDFConfiguration);
+
+        if (!lpg2rdfConfigs.hasNext()){
+            throw new IllegalArgumentException("LPGtoRDFConfiguration is required!");
+        }
+        else {
+            final Resource lpg2rdfConfig = lpg2rdfConfigs.next();
+
+            verifyLPG2RDFConfiguration(lpg2rdfConfig);
+
+            if(!lpg2rdfConfig.hasProperty(LPG2RDF.labelPredicate)){
+                throw new IllegalArgumentException("labelPredicate is required!");
+            }
+            final RDFNode labelPredicateObj = lpg2rdfConfig.getProperty(LPG2RDF.labelPredicate).getObject();
+            String url;
+            try {
+                url = labelPredicateObj.asResource().getURI();
+            }
+            catch (ResourceRequiredException re){
+                throw new IllegalArgumentException("labelPredicate is invalid!");
+            }
+            label = verifyExpectedURI(url).toString();
```

Comment:
Move the declaration of the `label` variable to here. I mean, this line should be:
```suggestion
            final String label = verifyExpectedURI(url).toString();
```

Reply:
Also, calling `verifyExpectedURI` here is not necessary. The label predicate URI in an LPG-to-RDF configuration can be an arbitrary URI; it does not need to be an HTTP or HTTPS URI.